src/hera.cpp

```diff
@@ -189,6 +229,38 @@ bytes evm2wasm(evmc_context* context, bytes_view input) {
   return ret;
 }
 
+// Calls the runevm contract.
+// @returns a wasm-based evm interpreter.
+bytes runevm(evmc_context* context, bytes code) {
+  HERA_DEBUG << "Calling runevm (code " << code.size() << " bytes)...\n";
+
+  int64_t gas = numeric_limits<int64_t>::max(); // do not charge for metering yet (give unlimited gas)
+  evmc_status_code status;
+  bytes ret;
+
+  tie(status, ret) = locallyExecuteSystemContract(
+      context,
+      runevmAddress,
+      gas,
+      {},
+      code,
+      code
+  );
+
+  HERA_DEBUG << "runevm done (output " << ret.size() << " bytes) with status=" << status << "\n";
+
+  heraAssert(ret.size() > 0, "Runevm returned empty.");
+  heraAssert(hasWasmPreamble(ret), "Runevm result has no wasm preamble.");
```

Comment:
Not entirely sure, but I think we should use `ensureCondition` here in both cases.